### ISSUE TITLE
Add click wikilink functionality?

### ISSUE THREAD

**Issue by quickfold** [kortina/vscode-markdown-notes#6]
I see that the intended link workflow is using peek and go to definition to look at linked notes. But is it possible to also allow navigation by clicking links, even if in the preview only? Currently, clicking the preview link tries to access the filename via browser as if it were a URL.

By the way, I recommend rewriting the documentation to make it clear help he can go to definition are supposed to work. It took me a while to figure it out. I'm happy to contribute on documentation if needed.

**Comment by jay-pee** [comment]
I made a short video for this problem:

![wikilinks](https://user-images.githubusercontent.com/9380980/78658490-7fbc5f80-78ca-11ea-8f5c-7e27b5a66021.gif)

I also think that the links in the preview should be clickable and don't send you to the browser.

> By the way, I recommend rewriting the documentation to make it clear help he can go to definition are supposed to work. It took me a while to figure it out. I'm happy to contribute on documentation if needed.

Thats the beauty of github!  You can download this repo  and integrate your changes to the markdown readme :). I think the author would love to see a pull request from you with a improved documentation!

You can just google yourself how to do it or use this link: 
https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github

**Comment by kortina** [comment]
I agree this would be sweet, but I'm not sure (1) if it is possible and (2) if so what APIs to use to do it. If anyone knows, please comment on this issue.

**Comment by lukesmurray** [comment]
You can use a [document link provider](https://code.visualstudio.com/api/references/vscode-api#DocumentLinkProvider) and fill in the provide document links function. The [document link](https://code.visualstudio.com/api/references/vscode-api#DocumentLink) has a target uri. I found that if you don't fill it in then it uses the definition provider. The nice side effect is if you auto create files in go to definition you can also auto create them when clicking on the document links.

This is what the links look like just from filling out the provider.

![image](https://user-images.githubusercontent.com/34020210/80931653-4839b980-8d89-11ea-94cc-675fdde76e84.png)

**Comment by lukesmurray** [comment]
Also this is for the editor. Not for the preview window. Getting this to work in the preview would probably require implementing your own function to convert markdown to html and overriding the existing preview.

**Comment by thomaskoppelaar** [comment]
@lukesmurray and @kortina I'm reviving this issue and adding this link: https://marketplace.visualstudio.com/items?itemName=thomaskoppelaar.markdown-wiki-links-preview

VSCode allows for markdown-it plugins to be added to the default preview window, which is something I'm taking advantage of in this case. 

Two caveats:
1. When you click a link in the preview, the editor won't switch files with it. e.g: if you click on `[[leaf]]` within `tree.md`'s preview, the preview will show `leaf.md`, whilst the editor stays on `branch.md`. This is something you can't fix unless you make use of a custom webview instead of the default preview. This is possible, but inadvisable.
2. The generated link won't look for files in subdirectories (you can point to them still): This could mean that markdown-notes recognizes that `[[leaf]]` points to `./branch/leaf.md`, whilst the preview extension will point to `./leaf.md`. A workaround of course is to use `[[branch/leaf.md]]`.

**Comment by kortina** [comment]
@thomaskoppelaar you are the maintainer of the preview extension, right? 

Have you seen the discussion of the API here:

https://github.com/kortina/vscode-markdown-notes/issues/71

This might be a good case for us to try out providing an API to the `Refs` found in the workspace by `NoteParser` (see implementation proposal towards the end of the issue description). You could use that to properly resolve to all the filenames instead of assuming workspace root as the path.

Lmk if you're interested in working on this together.

**Comment by thomaskoppelaar** [comment]
Sure, I think it'd be great to work together (or perhaps merge my extension into yours).

**Comment by denniseffing** [comment]
@kortina @thomaskoppelaar To be honest I'm quite confused. You fixed the rendering of wiki links in the markdown preview a while ago but navigating to wiki links using the preview still doesn't work for me, even when automatically generating link reference definitions in the file.

When the markdown notes plugin is **enabled**, clicking wiki links in the preview opens the browser with the following URL:
`https://file+.vscode-resource.vscode-webview.net/<file-path>` (actual behavior)

When the markdown notes plugin is **disabled**, clicking wiki links in the preview opens the local file in the preview instead (expected behaviour).

In an earlier comment you said that you don't know if this is possible at all. I suppose you meant that you don't know if it's possible to use links in the preview _without_ link reference definitions?

Would it be possible to _not break_ the functionality of link reference definitions in the preview when using markdown notes?
